distributed/protocol/core.py

```diff
@@ -104,7 +105,36 @@ def _encode_default(obj):
             else:
                 return msgpack_encode_default(obj)
 
-        frames[0] = msgpack.dumps(msg, default=_encode_default, use_bin_type=True)
+        try:
+            frames[0] = msgpack.dumps(msg, default=_encode_default, use_bin_type=True)
+        except TypeError as e:
+            logger.warning(e)
```

Comment:
I'm not sure how meaningful this is to users. Just logging the original exception appears to be an error for users but we're handling it gracefully. If anything we should log a custom message making them aware of a possible performance issue but I'm not sure if this is actually worth a warning log

Reply:
Fair enough, I'll de-escalate that to an INFO log with a custom message. I'd like to have something logged here to raise user attention and possibly for log analysis to highlight potential performance issues.